src/main/java/com/overops/plugins/service/impl/ReportBuilder.java

```diff
@@ -186,290 +189,198 @@ private Version createVersion(EventResult e) {
+				 .collect(Collectors.toList());
 		boolean hasRegressions = false;
-		if (runRegressions) {
-			rateRegression = RegressionUtil.calculateRateRegressions(apiClient, input, output, verbose);
-			
-			reportVolume = getReportVolume(apiClient, input, 
-					rateRegression, topEventLimit, regexFilter, output, verbose);
-				
-			regressions = getAllRegressions(apiClient, input, rateRegression, reportVolume.filter);
-			if (regressions != null && regressions.size() > 0) {
-				hasRegressions = true;
-				replaceSourceId2(regressions);
-			}
+
+		if (!regressionErrors.isEmpty()) {
+			hasRegressions = true;
 		}
-		
-		//max total error gate
-		boolean maxVolumeExceeded = (maxEventVolume != 0) && (qualityGateReport.getTotalErrorCount() > maxEventVolume);
-		
+
+		boolean maxVolumeExceeded = (properties.getMaxErrorVolume() != 0) && (totalErrorCount > properties.getMaxErrorVolume());
+
 		//max unique error gate
 		long uniqueEventCount;
 		boolean maxUniqueErrorsExceeded;
-		if (maxUniqueErrors != 0) {
-			uniqueEventCount = qualityGateReport.getUniqueErrorCount();
-			maxUniqueErrorsExceeded = uniqueEventCount > maxUniqueErrors;
+		if (properties.getMaxUniqueErrors() != 0) {
+			uniqueEventCount = uniqueErrorCount;
+			maxUniqueErrorsExceeded = uniqueEventCount > properties.getMaxUniqueErrors();
 		} else {
 			uniqueEventCount = 0;
 			maxUniqueErrorsExceeded = false;
 		}
-		
+
 		//new error gate
-		boolean newErrors = false;
-		if (qualityGateReport.getNewErrors() != null && qualityGateReport.getNewErrors().size() > 0) {
-			newErrors = true;
-			replaceSourceId(qualityGateReport.getNewErrors());
+		boolean newError = false;
+		if (newErrors.size() > 0) {
+			newError = true;
 		}
-		
+
 		//resurfaced error gate
 		boolean resurfaced = false;
-		if (qualityGateReport.getResurfacedErrors() != null && qualityGateReport.getResurfacedErrors().size() > 0) {
+		if (resurfacedErrors.size() > 0) {
 			resurfaced = true;
-			replaceSourceId(qualityGateReport.getResurfacedErrors());
 		}
-		
+
 		//critical error gate
 		boolean critical = false;
-		if (qualityGateReport.getCriticalErrors() != null  && qualityGateReport.getCriticalErrors().size() > 0) {
+		if (criticalErrors.size() > 0) {
 			critical = true;
-			replaceSourceId(qualityGateReport.getCriticalErrors());
 		}
-		
+
 		//top errors
-		if (qualityGateReport.getTopErrors() != null  && qualityGateReport.getTopErrors().size() > 0) {
-			replaceSourceId(qualityGateReport.getTopErrors());
+		if (topErrors.size() > 0) {
 		}
-		
+
 		boolean checkCritical = false;
-		if (input.criticalExceptionTypes != null && input.criticalExceptionTypes.size() > 0) {
+		if (input.regressionInput.criticalExceptionTypes != null && input.regressionInput.criticalExceptionTypes.size() > 0) {
 			checkCritical = true;
 		}
-		
+
 		boolean unstable = (hasRegressions)
 				|| (maxVolumeExceeded)
 				|| (maxUniqueErrorsExceeded)
-				|| (newErrors)
+				|| (newError)
```

Comment:
if you drop errors to error i would just change it to "new" and be done with it. but since that's not possible i guess adding "Events" to all is more pure